src/gui/log/logmodel.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Prince Gupta <jagannatharjun11@gmail.com>
+ * Copyright (C) 2019  sledgehammer999 <hammered999@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "logmodel.h"
+
+#include <QApplication>
+#include <QDateTime>
+#include <QColor>
+#include <QPalette>
+
+#include "base/global.h"
+#include "base/logger.h"
+
+BaseLogModel::BaseLogModel(QObject *object)
+    : QAbstractListModel(object)
+    , m_messages(MAX_LOG_MESSAGES)
```

Comment:
How about having less message count to be displayed and not follow Logger limit?
@Chocobo1?

Reply:
GUI widgets should have its own display limit.
Also I would prefer larger (around 50000) if it doesn't drag the performance.

Reply:
>Also I would prefer larger (around 50000) if it doesn't drag the performance.

Maybe it would be better to make it configurable?
Anyway the good first step is just declare another const for GUI stuff (even if it has the same value initially).

Reply:
>Maybe it would be better to make it configurable?

Maybe in a later PR to avoid delaying this one...

>Anyway the good first step is just declare another const for GUI stuff (even if it has the same value initially).

Agreed.

Reply:
What should be the const name? Can't think of one. MAX_GUI_LOG_MESSAGES?

Reply:
MAX_VISIBLE_MESSAGES?
Also seems Logger constant should be private.